### PR TITLE
Use enum values in Testing DataStore GetApprenticeshipsByIdsHandler

### PR DESCRIPTION
**Pull request by timabell**
Use enum values in Testing DataStore GetApprenticeshipsByIdsHandler

Just a suggestions, I might have missed a reason why we're not doing this.

### CODE REVIEW

File: tests/Dfc.CourseDirectory.Testing/DataStore/CosmosDb/QueryHandlers/GetApprenticeshipsByIdsHandler.cs [hunk 1]
```diff
@@ -12,7 +13,7 @@ public class GetApprenticeshipsByIdsHandler
         public IDictionary<Guid, Apprenticeship> Execute(InMemoryDocumentStore inMemoryDocumentStore, GetApprenticeshipsByIds request) =>
             inMemoryDocumentStore.Apprenticeships.All
                 .Where(p => request.ApprenticeshipIds.Contains(p.Id))
-                .Where(p => p.RecordStatus != 4 && p.RecordStatus != 8)
+                .Where(p => p.RecordStatus != (int)RecordStatus.Archived && p.RecordStatus != (int)RecordStatus.Deleted)
```
Comment (gunndabad) [inline tests/Dfc.CourseDirectory.Testing/DataStore/CosmosDb/QueryHandlers/GetApprenticeshipsByIdsHandler.cs:16]: Better to use `ApprenticeshipStatus` here; `RecordStatus` is from the old models and shouldn't be used going forward

File: tests/Dfc.CourseDirectory.Testing/Dfc.CourseDirectory.Testing.csproj [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\src\Dfc.CourseDirectory.Core\Dfc.CourseDirectory.Core.csproj" />
+    <ProjectReference Include="..\..\src\Dfc.CourseDirectory.Services\Dfc.CourseDirectory.Services.csproj" />
```
Comment (gunndabad) [inline tests/Dfc.CourseDirectory.Testing/Dfc.CourseDirectory.Testing.csproj:23]: Don't really want old stuff referenced from new stuff if we can avoid it.

File: tests/Dfc.CourseDirectory.Testing/DataStore/CosmosDb/QueryHandlers/GetApprenticeshipsByIdsHandler.cs (hunk 1 above)
Comment (timabell) [reply tests/Dfc.CourseDirectory.Testing/DataStore/CosmosDb/QueryHandlers/GetApprenticeshipsByIdsHandler.cs:16]: Ah, maybe we should use Obsolete to signpost to the new one, or at least a comment on the old one. Thanks

Comment (gunndabad) [reply tests/Dfc.CourseDirectory.Testing/DataStore/CosmosDb/QueryHandlers/GetApprenticeshipsByIdsHandler.cs:16]: `[Obsolete]` will fail build currently

Comment (timabell) [reply tests/Dfc.CourseDirectory.Testing/DataStore/CosmosDb/QueryHandlers/GetApprenticeshipsByIdsHandler.cs]: Yeah, changed my mind immediately, will also be very noise if we obsolete everything in legacy.

**Comment by timabell** [comment]
rebase + squash pushed....

**Pull request merged**